src/plugins/footnotes/index.ts

```diff
@@ -146,24 +145,35 @@ const deleteFootnoteWidget =
         if (node.type === schema.nodes.footnote && pos) {
           const targetNode = tableElement ? tableElement.node : view.state.doc
           const inlineFootnotes = getInlineFootnotes(id, targetNode)
-
-          const nodeWithPos = findParentNodeClosestToPos(
+          const footnotesElement = findParentNodeClosestToPos(
+            tr.doc.resolve(pos),
+            (node) => node.type === schema.nodes.footnotes_element
+          )
+          const tableElementFooter = findParentNodeClosestToPos(
             tr.doc.resolve(pos),
-            (node) => node.type === schema.nodes.footnote
+            (node) => node.type === schema.nodes.table_element_footer
           )
-          if (nodeWithPos) {
-            const { pos: fnPos, node: fnNode } = nodeWithPos
-            view.dispatch(tr.delete(fnPos, fnPos + fnNode.nodeSize))
+          // remove table-element-footer if it has only one footnote
+          if (
+            footnotesElement?.node.childCount === 1 &&
+            tableElementFooter?.node.childCount === 1
+          ) {
+            const { pos: fnPos, node: fnNode } = tableElementFooter
+            tr.delete(fnPos, fnPos + fnNode.nodeSize + 1)
+          } else {
+            const footnoteElement = findParentNodeClosestToPos(
+              tr.doc.resolve(pos),
+              (node) => node.type === schema.nodes.footnote
+            )
+            const { pos: fnPos, node: fnNode } = footnoteElement
+            tr.delete(fnPos, fnPos + fnNode.nodeSize + 1)
           }
 
           // delete inline footnotes
           if (inlineFootnotes) {
-            tr = view.state.tr
```

Comment:
is there any reason for removing this ?
I believe we had to do this before to cover very accurate failed cases that affect the position of the footnote

Reply:
yes, because the error mean was occurring, because of the logic in the following file https://github.com/Atypon-OpenSource/manuscripts-body-editor/blob/f3541e98b0c02c68ba9b8c1da44e9db94ae7ad08/src/plugins/table-footnote.ts#L84

and re-declare the tr will ignore any previously attached actions.

Reply:
actually its related to the following line:
https://github.com/Atypon-OpenSource/manuscripts-body-editor/blob/5dd87dbf5e82a7936fda79345bc3659691fa5be5/src/plugins/footnotes/index.ts#L181